src/components/RefinementList/RefinementList.js

```diff
@@ -146,65 +144,102 @@ export class RawRefinementList extends Component {
+      [`${this.props.cssClasses.depth}${this.props.depth}`]: this.props
+        .cssClasses.depth,
+    });
+
+    const showMoreButtonClassName =
+      this.props.showMore === true && this.props.canToggleShowMore
+        ? this.props.cssClasses.showMore
+        : cx(
+            this.props.cssClasses.showMore,
+            this.props.cssClasses.disabledShowMore
+          );
 
-    const showMoreBtn =
-      this.props.showMore === true && this.props.canToggleShowMore ? (
+    const showMoreButton =
+      this.props.showMore === true ? (
         <Template
-          rootProps={{ onClick: this.props.toggleShowMore }}
+          rootTagName="button"
           templateKey={`show-more-${
             this.props.isShowingMore ? 'active' : 'inactive'
           }`}
+          rootProps={{
+            className: showMoreButtonClassName,
+            onClick: this.props.toggleShowMore,
+          }}
           {...this.props.templateProps}
         />
-      ) : (
-        undefined
-      );
+      ) : null;
 
-    const shouldDisableSearchInput =
+    const shouldDisableSearchBox =
       this.props.searchIsAlwaysActive !== true &&
       !(this.props.isFromSearch || !this.props.hasExhaustiveItems);
-    const searchInput = this.props.searchFacetValues ? (
-      <SearchBox
-        ref={i => {
-          this.searchbox = i;
-        }}
-        placeholder={this.props.searchPlaceholder}
-        onChange={this.props.searchFacetValues}
-        onValidate={() => this.refineFirstValue()}
-        disabled={shouldDisableSearchInput}
-      />
+    const searchBox = this.props.searchFacetValues ? (
+      <div className={this.props.cssClasses.searchBox}>
+        <SearchBox
+          ref={i => {
+            this.searchbox = i;
+          }}
+          placeholder={this.props.searchPlaceholder}
+          onChange={this.props.searchFacetValues}
+          onValidate={() => this.refineFirstValue()}
+          disabled={shouldDisableSearchBox}
+        />
+      </div>
     ) : null;
 
+    const facetValues =
+      this.props.facetValues && this.props.facetValues.length > 0 ? (
+        <ul className={cssClassList}>
+          {this.props.facetValues.map(this._generateFacetItem, this)}
+        </ul>
+      ) : null;
+
     const noResults =
       this.props.searchFacetValues &&
       this.props.isFromSearch &&
       this.props.facetValues.length === 0 ? (
-        <Template templateKey={'noResults'} {...this.props.templateProps} />
+        <Template
+          templateKey="noResults"
+          rootProps={{ className: this.props.cssClasses.noResults }}
+          {...this.props.templateProps}
+        />
       ) : null;
 
     return (
-      <div className={cx(cssClassList)}>
-        {searchInput}
-        {this.props.facetValues.map(this._generateFacetItem, this)}
+      <div
+        className={cx(this.props.cssClasses.root, {
+          [this.props.cssClasses.noRefinementRoot]: false,
```

Comment:
What is the condition for a refinement list to be refined?

Reply:
when there's no possible values I think